packages/middleware-flexible-checksums/src/validateChecksumFromResponse.ts

```diff
@@ -29,9 +32,20 @@ export const validateChecksumFromResponse = async (
     const checksumFromResponse = responseHeaders[responseHeader];
     if (checksumFromResponse) {
       const checksumAlgorithmFn = selectChecksumAlgorithmFunction(algorithm as ChecksumAlgorithm, config);
-      const { streamHasher, base64Encoder } = config;
-      const checksum = await getChecksum(responseBody, { streamHasher, checksumAlgorithmFn, base64Encoder });
+      const { base64Encoder } = config;
 
+      if (isStreaming(responseBody)) {
+        createChecksumStream({
```

Comment:
Isn't the return value of this needed?

Reply:
The assert in tests show that body is getting transformed, and timing indicates checksum validation is happening, although it's not verified. Is it because the Checksum Duplex Stream is created, but is not consumed?

I'll assign the return value to body, since that's how the Duplex stream is designed to be used.

Reply:
response.body was updated in [`e3d92aa` (#6629)](https://github.com/aws/aws-sdk-js-v3/pull/6629/commits/e3d92aada9238d90045ca0c9746509312334394c)